dbms/src/Flash/Pipeline/Schedule/Events/Event.cpp

```diff
@@ -38,6 +38,12 @@ extern const char random_pipeline_model_event_finish_failpoint[];
         exec_status.onErrorOccurred(std::current_exception());   \
     }
 
+Event::Event(PipelineExecutorStatus & exec_status_, const String & req_id)
+    : exec_status(exec_status_)
+    , mem_tracker(exec_status_.getMemoryTracker())
```

Comment:
Seems `exec_status` is hold by `Event`, and `exec_status` already holds `mem_tracker`, why need to hold the `mem_tracker` again in `Event`?

Reply:
Because when mpptask is destructed, event/task may not be destructed yet.
When the ref count of exec_status change to 0, the mpptask can end, but this does not mean that the event/task are all destructed

Reply:
However, as long as we ensure that the MemoryTracker is not called after exec_status.decRefCount is invoked, we can eliminate the need for the event to hold a MemoryTrackerPtr.

Reply:
Because after calling exec_context.decActiveRefCount, the memory tracker is still referenced by MemoryTrackerSetter, so an additional MemoryTrackerSharedPtr is needed here.
```
void Event::finish()
{
    switchStatus(EventStatus::SCHEDULED, EventStatus::FINISHED);
    finish_duration = stopwatch.elapsed();
    MemoryTrackerSetter setter{true, exec_context_.getMemoryTracker().get()};
   ...
    exec_context.decActiveRefCount();
}
```

Reply:
OK